sdk/src/binary.rs

```diff
@@ -403,6 +403,25 @@ impl<const N: u32> FixedLengthBinary for ArrayBinary<N> {
     }
 }
 
+impl<const N: usize, const M: u32> TryFrom<EnvType<[u8; N]>> for ArrayBinary<M> {
```

Comment:
This probably looks a bit odd, and we probably need a better name for `EnvType`, but what this conversion is saying is that you can convert from an `[u8; N]` to an `ArrayBinary<N>` if you also have an `Env` present.